internal/provider/data_source_http.go

```diff
@@ -55,6 +58,22 @@ your control should be treated as untrustworthy.`,
 				Computed:    true,
 			},
 
+			"ca_certificate": {
+				Description: "PEM-encoded root certificate bundle for TLS authentication",
+				Type:        types.StringType,
+				Optional:    true,
+			},
+
+			"insecure": {
+				Description: "Whether server should be accessed without verifying the TLS certificate. Defaults to false",
+				Type:        types.BoolType,
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
```

Comment:
PlanModifiers are ignored if they are present on attributes for data sources or providers (see [docs](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework/tfsdk#Attribute)).
```
// Plan modification only applies to resources, not data sources or
// providers. Setting PlanModifiers on a data source or provider attribute
// will have no effect.
```